src/decorators/CustomField.ts

```diff
@@ -27,18 +27,60 @@ interface CustomFieldOptions {
 export function CustomField(args: CustomFieldOptions): any {
   // const nullableOption = typeof args.nullable !== 'undefined' ? { nullable: args.nullable } : {};
   // const dbOptions = { ...nullableOption, ...(args.db || {}) };
-  const { type, filter, sort, ...typeGraphQLOptions } = args.api;
-  const warthogOptions = { nullable: args.api.nullable, type, filter, sort, array: args.db.array };
+  const {
+    type,
+    filter,
+    sort,
+    readonly,
+    apiOnly,
+    editable,
+    dbOnly,
+    writeonly,
+    ...typeGraphQLOptions
+  } = args.api;
+  const warthogOptions: any = {
+    nullable: args.api.nullable,
+    type,
+    filter,
+    sort,
+    array: args.db.array,
+    readonly,
+    apiOnly,
+    editable,
+    dbOnly,
+    writeonly
+  };
   const graphQLType = args.db.array
     ? [columnTypeToGraphQLType(args.api.type)]
     : columnTypeToGraphQLType(args.api.type);
 
-  // These are the 2 required decorators to get type-graphql and typeorm working
-  const factories = [
-    WarthogField(args.api.type, warthogOptions),
-    Field(() => graphQLType, typeGraphQLOptions),
-    Column(args.db) as MethodDecoratorFactory
-  ];
+  const factories: any[] = [];
+
+  const exposeDB = dbOnly || !apiOnly;
+  const exposeAPI = apiOnly || !dbOnly;
+
+  // Warthog: start with the Warthog decorator that adds metadata for generating the GraphQL schema
+  // for sorting, filtering, args, where inputs, etc...
+  if (exposeAPI) {
+    // filter out null and undefined from the list so they get proper defaults
+    const filteredOptions: any = Object.entries(warthogOptions).reduce(
+      (a: any, [k, v]) => (v == null ? a : ((a[k] = v), a)),
+      {}
+    );
+
+    factories.push(WarthogField(type, filteredOptions));
+  }
```

Comment:
We need to filter out null and undefined from this object because unlike a more traditional case like `StringField` where this config object is just passed in to the decorator and used we are building it from different parts of `api` and `db`.  By specifying the fields directly on this object if they are null or undefined in the original object they will be explicitly put on the resulting `warthogOptions` object as null or undefined (vs being implied as such).  This means we override our defaults in `composeMethodDecorators` even though we want to use those defaults.